AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomGPU_BasicLevelSetup.py

```diff
@@ -98,193 +137,192 @@ def AtomGPU_BasicLevelSetup_SetsUpLevel():
     def initial_viewport_setup(screen_width, screen_height):
         general.set_viewport_size(screen_width, screen_height)
         general.update_viewport()
-        result = isclose(
-            a=general.get_viewport_size().x, b=SCREEN_WIDTH, rel_tol=0.1) and isclose(
-            a=general.get_viewport_size().y, b=SCREEN_HEIGHT, rel_tol=0.1)
-
-        return result
+        TestHelper.wait_for_condition(
+            function=lambda: isclose(a=general.get_viewport_size().x, b=SCREEN_WIDTH, rel_tol=0.1)
+                        and isclose(a=general.get_viewport_size().y, b=SCREEN_HEIGHT, rel_tol=0.1),
+            timeout_in_seconds=4.0
+        )
 
     with Tracer() as error_tracer:
         # Test setup begins.
         # Setup: Wait for Editor idle loop before executing Python hydra scripts then open "Base" level.
-        helper.init_idle()
-        helper.open_level("", "Base")
+        TestHelper.init_idle()
+        TestHelper.open_level("", "Base")
 
         # Test steps begin.
         # 1. Close error windows and display helpers then update the viewport size.
-        helper.close_error_windows()
-        helper.close_display_helpers()
+        TestHelper.close_error_windows()
+        TestHelper.close_display_helpers()
+        initial_viewport_setup(SCREEN_WIDTH, SCREEN_HEIGHT)
         general.update_viewport()
-        Report.critical_result(Tests.viewport_set, initial_viewport_setup(SCREEN_WIDTH, SCREEN_HEIGHT))
 
         # 2. Create Default Level Entity.
         default_level_entity_name = "Default Level"
         default_level_entity = EditorEntity.create_editor_entity_at(
             math.Vector3(0.0, 0.0, 0.0), default_level_entity_name)
 
         # 3. Create Grid Entity as a child entity of the Default Level Entity.
-        grid_name = "Grid"
-        grid_entity = EditorEntity.create_editor_entity(grid_name, default_level_entity.id)
+        grid_entity = EditorEntity.create_editor_entity(AtomComponentProperties.grid(), default_level_entity.id)
 
         # 4. Add Grid component to Grid Entity and set Secondary Grid Spacing.
-        grid_component = grid_entity.add_component(grid_name)
-        secondary_grid_spacing_property = "Controller|Configuration|Secondary Grid Spacing"
+        grid_component = grid_entity.add_component(AtomComponentProperties.grid())
+        secondary_grid_spacing_property = AtomComponentProperties.grid('Secondary Grid Spacing')
         secondary_grid_spacing_value = 1.0
         grid_component.set_component_property_value(secondary_grid_spacing_property, secondary_grid_spacing_value)
         secondary_grid_spacing_set = grid_component.get_component_property_value(
             secondary_grid_spacing_property) == secondary_grid_spacing_value
```

Comment:
why not use AtomComponentProperties.grid('Secondary Grid Spacing directly where secondary_grid_spacing_property is used?